src/main/java/com/toptrumps/db/PersistentDataDAOImpl.java

```diff
@@ -0,0 +1,110 @@
+package com.toptrumps.db;
+import java.sql.*; 
+import java.sql.Connection; 
+import java.sql.ResultSet; 
+import java.sql.SQLException; 
+import java.sql.Statement; 
+
+public class PersistentDataDAOImpl implements PersistentDataDAO{
+
+    public PersistentDataDAOImpl(){
+    }
+
+    public void update(){
+        
+        Connection con = ConnectionFactory.getConnection();
+        PreparedStatement updateBasicData = null;
+        PreparedStatement updateAiWins = null;
+        PreparedStatement updateHumanWins = null;
+
+        String updateBasicDataSQL = 
+
+        "UPDATE persistent_game_data " +  
+                "SET games_played = subquery.games_played, " +  
+                "draws_average = subquery.draws_average, " +   
+                "max_rounds = subquery.max_rounds " +  
+                "FROM (SELECT COUNT (*) AS games_played, " +  
+                "  AVG(draws) AS draws_average, " +  
+                "  MAX(rounds_played) AS max_rounds " +   
+                "  FROM  individual_game_data) " + 
+                "  AS subquery; ";
+
+        String updateAiWinsSQL = 
+
+        "UPDATE persistent_game_data " +
+                "SET ai_wins = subquery.ai_wins " +  
+                "FROM (SELECT COUNT (winner) AS ai_wins " +  
+                "  FROM individual_game_data " +
+                "  WHERE (winner != 'Human')) " +  
+                "  AS subquery; ";
+
+        String updateHumanWinsSQL =
+
+        "UPDATE persistent_game_data " +  
+        "SET human_wins = subquery.human_wins " + 
+        "FROM (SELECT COUNT (winner) AS human_wins " +  
+        "  FROM individual_game_data " + 
+        "  WHERE (winner = 'Human')) " +  
+        "  AS subquery; ";
```

Comment:
Inconsistent indentation - please amend

Reply:
Indentation is for the subquery in the query, but i can probably find a better way to do this